packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts

```diff
@@ -217,6 +232,10 @@ function patchedObserve(
         defineProperty(target, observerLookupField, { value: [] });
     }
     ArrayPush.call(target[observerLookupField], this);
+    if (isUndefined(this.observerLookups)) {
+        this.observerLookups = [];
```

Comment:
it is too risky to use an expando on the observer, I will be ok with a symbol, but I prefer a weakmap.

Reply:
but that's a question for @ravijayaramappa since he implemented the previous version.

Reply:
independently of what it is, it should never be undefined, it should always be an array so we avoid having to do forking logic in different places.

Reply:
to be clear... my problem here is that `wrapperLookupField` is used on a function that you hand over to the observer, which is fine... because you own/or have access to the observer, while `observerLookupField` is placed on the node itself, and the node is accessible to anyone with access to the DOM... that's why they are different in nature.